src/Repository/ArticleToCategory/Postgresql/PostgresqlCollection.php

```diff
@@ -0,0 +1,52 @@
+<?php
+declare(strict_types=1);
+
+namespace Repository\ArticleToCategory\Postgresql;
+
+class PostgresqlCollection extends \Maleficarum\Storage\Repository\Postgresql\Collection {
+    /**
+     * Return shard
+     *
+     * @param $collection
+     *
+     * @return \Maleficarum\Storage\Shard\ShardInterface
+     */
+    public function getShard($collection) {
+        return $this->getStorage()->fetchShard('Postgresql', ($this->shardSelector)($collection));
+    }
+
+    /**
+     * @see \Maleficarum\Storage\Repository\CollectionInterface.deleteAll()
+     */
+    public function deleteAll(\Maleficarum\Data\Collection\Persistable\AbstractCollection $collection): \Maleficarum\Storage\Repository\CollectionInterface {
+        $shard = $this->getShard($collection);
+
+        // Extract entity ids
+        $articleIds = [];
+        $categoryIds = [];
+        foreach ($collection->toArray() as $key => $col_element) {
+            isset($col_element['mapArticleCategoryArticleId']) and $articleIds[':_id_token_articleId_'.$key] = $col_element['mapArticleCategoryArticleId'];
+            isset($col_element['mapArticleCategoryCategoryId']) and $categoryIds[':_id_token_categoryId_'.$key] = $col_element['mapArticleCategoryCategoryId'];
+        }
+
+        // Build the sql query
+        $sql = 'DELETE FROM "' . $collection->getStorageGroup() . '" WHERE ';
+        $limit = \count($articleIds);
+        for ($i=0; $i<$limit; $i++){
```

Comment:
ten zabieg z `for` przy tworzeniu SQLki jest bardzo nieczytelny i imo błędogenny. Nie można było najpierw utworzyć tablic articleIds oraz categoryIds a potem skorzystać z `IN` w SQLu i przekazać mu tablice zamiast się tak brzydko eachować?